core/server/api/schedules.js

```diff
@@ -6,25 +6,39 @@ var _ = require('lodash'),
     dataProvider = require(config.paths.corePath + '/server/models'),
     i18n = require(config.paths.corePath + '/server/i18n'),
     errors = require(config.paths.corePath + '/server/errors'),
+    apiPosts = require(config.paths.corePath + '/server/api/posts'),
     utils = require('./utils');
 
 /**
  * publish a scheduled post
  */
 exports.publishPost = function publishPost(options) {
+    var post = null;
     options = options || {};
 
     // CASE: only the scheduler client is allowed to publish (hardcoded because of missing client permission system)
     if (!options.context || !options.context.client || options.context.client !== 'ghost-scheduler') {
         return Promise.reject(new errors.NoPermissionError(i18n.t('errors.permissions.noPermissionToAction')));
     }
 
+    options.context = {internal: true};
+
     return pipeline([
         utils.validate('posts', {opts: utils.idDefaultOptions}),
         function (cleanOptions) {
-            return dataProvider.Post.edit({status: 'published'}, _.omit(cleanOptions, 'data'))
+            cleanOptions.status = 'scheduled';
+
+            return apiPosts.read(cleanOptions)
                 .then(function (result) {
-                    return Promise.resolve({post: result.toJSON(cleanOptions)});
+                    post = result.posts[0];
+
+                    // CASE: now and published_at should almost be equal (1minute tolerance)
```

Comment:
That is a good use case (scheduler is down), but talking about timezones, what if somebody configures the TZ not to UTC, instead to UTC-6, the post would get scheduled 6 hours too early?

@ 2mins: sure we can use the the same minute offset for both cases 👍